### ISSUE TITLE
Remember order of tags in BibTeX entry

### ISSUE THREAD

**Issue by pjhuxford** [charlesvdv/nom-bibtex#28]
Currently the tags of a bibliography entry are stored in a `HashMap`, which means the order of the tags is unstable.

I have a use-case in mind where I would like to parse a bibliography entry from a file, perform some minor edits, and then write the new entry back to the same file. If I use `nom-bibtex` for the parsing then I have no way to recover the original order of the tags in the entry. This could potentially make the version control history of the file confusing to decipher.

If you're also interested in remembering the order of the tags, I suppose one option would be to replace `HashMap` with `IndexMap` from the [indexmap](https://crates.io/crates/indexmap) crate. Another option would be storing a `Vec<String>` of the keys alongside the `HashMap`.

**Comment by A6GibKm** [comment]
Makes sense to me, but returning a struct not in std is kinda annoying.

**Comment by pjhuxford** [comment]
Yeah it is annoying that `IndexMap` isn't in std, although it does seem like the most performant way to handle this. Maybe storing the keys of the tags in a `Vec<String>` is a better approach though. What do you think?
